src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -372,12 +372,13 @@ private function parseGroups(string $regex): ?array
 	{
 		if (self::$parser === null) {
 			/** @throws void */
-			self::$parser = Llk::load(new Read('hoa://Library/Regex/Grammar.pp'));
+			self::$parser = Llk::load(new Read(__DIR__ . '/../../../resources/RegexGrammar.pp'));
 		}
 
 		try {
 			$ast = self::$parser->parse($regex);
```

Comment:
@staabm another thing I noticed while working on this, it looks like this parses every regex several times, it might be worth memoizing the result to save cpu cycles. Not sure but maybe worth considering.

Reply:
I had the same idea today and did some profiles. the parsing is consuming quite a bit, but I was not yet able to find a real world project in which this has a impact on overall scan time